File: mqtt_panel/web/screenoverlay.js
class ScreenOverlay {
    constructor() {
        var This = this;
        $('#screen-overlay').on('alert', function(event, message, onclick) {
            This.alert(message, onclick);
        });
        $('#screen-overlay').on('spinner', function(event) {
            This.spinner();
        });
        $('#screen-overlay').on('hide', function(event) {
            This.hide();
        });
        this.onclick = undefined;
        $('#screen-overlay .alert').on('click', function(event){
            if (This.onclick !== undefined) {
                This.onclick();
            }
            return false;
        });
    }

    alert(message, onclick) {
        $('#screen-overlay').removeClass('d-none');
        $('#screen-overlay .alert').removeClass('d-none');
        $('#screen-overlay .alert').html(message);
        this.onclick = onclick;
    }

    spinner() {
        $('#screen-overlay').removeClass('d-none');
        $('#screen-overlay .spinner').removeClass('d-none');
    }    

    hide() {
        $('#screen-overlay').addClass('d-none');
        $('#screen-overlay .alert').addClass('d-none');
        $('#screen-overlay .spinner').addClass('d-none');
    }
}

File: mqtt_panel/web/screenoverlay.py
from mqtt_panel.web.component import Component


class ScreenOverlay(Component):
    def __init__(self):
        super().__init__(4)

    def _body(self, fh):
        self._write_render(fh, '''\
            <div id="screen-overlay" class="d-none">
                <div class="alert d-none">Alert</div>
                <div class="spinner d-none"></div>
            </div>
        ''', indent=self._indent)
